src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSetEqualityComparer.cs

```diff
@@ -32,27 +23,22 @@ public int GetHashCode(HashSet<T>? obj)
                 {
                     if (t != null)
                     {
-                        hashCode ^= (_comparer.GetHashCode(t) & 0x7FFFFFFF);
+                        hashCode ^= _comparer.GetHashCode(t) & 0x7FFFFFFF;
```

Comment:
Why are we still masking the hashcode? That was only necessary because we used the high bit to store whether the Entry was "empty". With this change, HashSet follows the Dictionary plan of using `Entry.next` to encode that, allowing it to preserve the extra bit of entropy.

Reply:
Scratch that, this is for some different purpose.. do you know what it is?

Reply:
This is likely left-over from one of the attempts to make tests pass with `/checked+`. I think it should be fine to remove the mask.

Reply:
`SortedSetEqualityComparer` has the same oddity.

Reply:
I think `/checked+` does not affect the `^` operation.  We have had this masking since the original implementation in 2006 — probably copied from the `Dictionary` implementation.